### PR TITLE
Cleanup date-like types

### LINKED ISSUE #81: Remove `Date` and `Time` types
Superseded by `DateOnly` i `TimeOnly`

### PR DESCRIPTION
**Pull request by jakubfijalkowski**
Draft, since it targets PR #90 .

Closes #81

### CODE REVIEW

**Comment by Saancreed** [comment]
:+1: on removal of `Date` and `Time,` :-1: on removal of `DateTime`. I'd like to talk about this during our weekly today.

**Review by lukaszgarstecki — APPROVED** [review]

**Review by Saancreed — APPROVED** [review]

**Comment by jakubfijalkowski** [comment]
@Saancreed I've added docs that describe why `DateTime` got removed. Please verify if the explanation is OK.

File: docs/dates.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Dates in contracts
+
+The contracts provide 4 date/datetime-like known types:
+
+1. `DateOnly` - represents a _date_ **without** time, e.g. 2022-09-08,
+2. `TimeOnly` - represents a time of day, e.g. 16:02:20. It is limited to 23:59:59.999,
+3. `DateTimeOffset` - represents date with time and offset (e.g. event in a time and space), e.g. `2022-09-08 16:02:20 +02:00`,
+4. `TimeSpan` - represents a duration of time, e.g. `2.10:50:30.222` (2 days, 10 hours, 50 min, 30 sec and 222 ms).
```
Comment (Saancreed) [inline docs/dates.md:8]: I'm not sure if this is mean to give examples of how it's supposed to look in serialized form / over the wire. All but DTO one look like this. Intended (and DTO should have spaces removed/replaced) or by accident?

**Review by Saancreed — APPROVED** [review]
LGTM but I have one question about a small bit of the docs that could be considered misleading.

File: docs/dates.md (hunk 1 above)
Comment (jakubfijalkowski) [reply docs/dates.md:8]: It's meant to represent meaning, not format. I probably should remove the code formatting.

**Review by shilangyu — APPROVED** [review]

**Review by konowrockis — APPROVED** [review]

**Pull request merged**